### PR TITLE
Setup snippet compiler for scala.quoted and fix snippets

### PR DESCRIPTION
**Pull request by KacperFKorban**

### CODE REVIEW

File: library/src/scala/quoted/Type.scala [hunk 1]
```diff
@@ -42,11 +51,20 @@ object Type:
...
+   *  //{
+   *  import scala.deriving.*
+   *  def f(using Quotes) = {
+   *  import quotes.reflect.*
+   *  val expr: Expr[Any] = ???
+   *  //}
+   *  expr match {
    *    case '{ $mirrorExpr : Mirror.Sum { type MirroredElemLabels = label } } =>
    *      Type.valueOfTuple[label] // Option[Tuple]
    *  }
```
Comment (KacperFKorban) [inline library/src/scala/quoted/Type.scala:64]: @nicolasstucki
This snippet fails with `Type argument label does not conform to upper bound Tuple`.
Is this a bug?

**Review by pikinier20 — APPROVED** [review]
For me, it looks OK. However, it should be seen by someone with more knowledge about `scala.quoted` package.

File: library/src/scala/quoted/Type.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src/scala/quoted/Type.scala:64]: Yes, it is a bug.

```diff
-    case '{ $mirrorExpr : Mirror.Sum { type MirroredElemLabels = label } } =>
+    case '{ type label <: Tuple; $mirrorExpr : Mirror.Sum { type MirroredElemLabels = `label` } } =>

```

Comment (KacperFKorban) [reply library/src/scala/quoted/Type.scala]: Ok, changed here. Also created #13527

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**